projects/whats_cooking_good_looking/whats_cooking_good_looking/train_ner_workflow.py

```diff
@@ -0,0 +1,261 @@
+import json
+import pickle
+import random
+from collections import defaultdict
+from pathlib import Path
+
+import spacy
+from flytekit import Resources, dynamic, task, workflow
+from spacy.language import Language
+from spacy.training import Example
+from spacy.util import compounding, minibatch
+
+from whats_cooking_good_looking.utils import (download_bytes_from_gcs,
+                                              download_from_gcs, load_config,
+                                              upload_to_gcs)
+
+SPACY_MODEL = {"en": "en_core_web_sm"}
+
+CACHE_VERSION = "2.2"
+request_resources = Resources(cpu="1", mem="500Mi", storage="500Mi")
+limit_resources = Resources(cpu="2", mem="1000Mi", storage="1000Mi")
+
+THRESHOLD_ACCURACY = 0.7
+
+
+@task
+def evaluate_ner(labelstudio_tasks: bytes) -> dict:
+    """Computes accuracy, precision and recall of NER model out of label studio output.
+
+    Args:
+        labelstudio_tasks (list): List of dicts outputs of label studio annotation with following format
+                        [
+                            {
+                            "result": [
+                                {
+                                    "value": {"start": 10, "end": 17, "text": "Chennai", "labels": ["LOC"]},
+                                    "from_name": "label",
+                                    "to_name": "text",
+                                    "type": "labels",
+                                    "origin": "manual",
+                                }
+                            ],
+                            "predictions": [
+                                    {
+                                    "result": {"start": 10, "end": 17, "text": "Chennai", "labels": ["LOC"]},
+                                    "model_version": "dummy",
+                                    }
+                                ],
+                            }
+                        ]
```

Comment:
Are you modifying the `predictions` format when the labels are manually assigned? Here's the format I see after running `apply_ner_workflow` code:

```
[
  {
    "data": {
      "text": "@o_Spitfire_o @DestinyTheGame And? You have chosen OPTIONAL content in a time frame when said content IS NOT ACCESIBLE. No surprise there are no metrics for those. It's not hard, dude, come on.  It's like measuring COD numbers based on one gun skin that people can't even get now."
    },
    "predictions": [
      {
        "model_version": "en_core_web_sm",
        "result": [
          {
            "from_name": "label",
            "to_name": "text",
            "type": "labels",
            "value": {
              "start": 51,
              "end": 59,
              "text": "OPTIONAL",
              "labels": [
                "ORG"
              ]
            }
          },
          {
            "from_name": "label",
            "to_name": "text",
            "type": "labels",
            "value": {
              "start": 215,
              "end": 218,
              "text": "COD",
              "labels": [
                "ORG"
              ]
            }
          },
          {
            "from_name": "label",
            "to_name": "text",
            "type": "labels",
            "value": {
              "start": 236,
              "end": 239,
              "text": "one",
              "labels": [
                "CARDINAL"
              ]
            }
          }
        ]
      }
    ]
  },
...
]
```

Reply:
Yes the difference is normal. The`apply_ner_workflow ` pipeline outputs a file in a format that can be ingested by our labelling interface (Label Studio). Then, a person manually annotates those observations in Label Studio and outputs another file (including their annotations) that has the format specified above and that is taken as input of the `train_ner_workflow `. Please let me know if my explications are clear enough.